src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1272,3 +1272,109 @@ rpmostreed_transaction_new_cleanup (GDBusMethodInvocation *invocation,
   return (RpmostreedTransaction *) self;
 
 }
+
+/* ================================ MakeCache ================================ */
+
+typedef struct {
+  RpmostreedTransaction parent;
+  char *osname;
+} MakeCacheTransaction;
+
+typedef RpmostreedTransactionClass MakeCacheTransactionClass;
+
+GType makecache_transaction_get_type (void);
+
+G_DEFINE_TYPE (MakeCacheTransaction,
+               makecache_transaction,
+               RPMOSTREED_TYPE_TRANSACTION)
+
+static void
+makecache_transaction_finalize (GObject *object)
+{
+  MakeCacheTransaction *self;
+
+  self = (MakeCacheTransaction *) object;
+  g_free (self->osname);
+
+  G_OBJECT_CLASS (makecache_transaction_parent_class)->finalize (object);
+}
+
+static gboolean
+makecache_transaction_execute (RpmostreedTransaction *transaction,
+                                GCancellable *cancellable,
+                                GError **error)
+{
+  MakeCacheTransaction *self = (MakeCacheTransaction *) transaction;
+  OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
+  g_autoptr(OstreeRepo) repo = NULL;
+  if (!ostree_sysroot_get_repo (sysroot, &repo, cancellable, error))
```

Comment:
In new code this can just be:
`OstreeRepo *repo = ostree_sysroot_repo (sysroot);`.